src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1630,6 +1619,10 @@ void TorrentHandle::handleMetadataReceivedAlert(libt::metadata_received_alert *p
         m_speedMonitor.reset();
         m_session->handleTorrentPaused(this);
     }
+    
+    // If first/last piece priority was specified when adding this torrent, set it up now that we can:
+    if (m_hasFirstLastPiecePriority)
+        setFirstLastPiecePriority(true);
```

Comment:
I agree. 
This was done almost the same until the author decided to get into the inside.

>Update: the flag has to be serialised always, of course, otherwise we would read it as set from the resume file always.

Disagree. We should write this flag to resume data only if torrent hasn't metadata (as we do for addPaused/addForced).